### PR TITLE
add EraSearch Vector Sink

### PR DESCRIPTION
**Pull request by tebbers**
This pull request will add the EraSerach vector sink as well as update the readme and the shell script.

### CODE REVIEW

File: vector-configs/sinks/erasearch.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[sinks.erasearch]
+  type="elasticsearch
```
Comment (jbergstroem) [inline vector-configs/sinks/erasearch.toml:2]:
```suggestion
  type="elasticsearch"
```

**Comment by tebbers** [comment]
Thanks @jbergstroem Sorry i missed that one.  My cut and paste failed me. :). Take a look at the .sh.  Not sure if that's what you were looking for, but tried to follow the pattern you started.  I was wondering if all of the variables should be there or not.

**Comment by tebbers** [comment]
I also noticed that I didn't save my updated TOML.  Did all uppercase to follow the pattern.  Sorry I missed that.

**Comment by jbergstroem** [comment]
(I'm not an contributor to this repo, just noticed a typo!)

File: README.md [hunk 2]
```diff
@@ -82,6 +82,14 @@ One of these is required for New Relic logs. New Relic recommend the license key
 | `UPTRACE_API_KEY` | Uptrace API key    |
 | `UPTRACE_PROJECT` | Uptrace project ID |
 
+### EraSearch
+
+| Secret                  | Description                                  |
+| ----------------------- | -------------------------------------------- |
+| `ERASERACH_URL`         | EraSearch Endpoint                           |
```
Comment (sstangl) [inline README.md:89]: `s/ERASERACH/ERASEARCH/g` in this file.

Also `s/EraSerach/EraSearch/g` for the comments.

File: start-fly-log-transporter.sh [hunk 3]
```diff
@@ -15,6 +15,7 @@ fi
 [[ ! -z "$PAPERTRAIL_ENDPOINT" ]] && cat /etc/vector/papertrail.toml >> /etc/vector/vector.toml 
 [[ ! -z "$SEMATEXT_TOKEN" ]] && cat /etc/vector/sematext.toml >> /etc/vector/vector.toml
 [[ ! -z "$UPTRACE_API_KEY" ]] && [[ ! -z "$UPTRACE_PROJECT" ]] && cat /etc/vector/uptrace.toml >> /etc/vector/vector.toml
+[[ ! -z "$ERASERACH_URL" ]] && [[ ! -z "$ERASERACH_INDEX" ]] && cat /etc/vector/eraserach.toml >> /etc/vector/vector.toml
```
Comment (sstangl) [inline start-fly-log-transporter.sh:18]: `s/ERA_SERACH/ERA_SEARCH/g` and `s/eraserach/erasearch`.

File: vector-configs/sinks/erasearch.toml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+[sinks.erasearch]
+  type="elasticsearch"
+  inputs = ["log_json"]
+  endpoint="${ERASERACH_URL}"
```
Comment (sstangl) [inline vector-configs/sinks/erasearch.toml:4]: `s/ERASERACH/ERASEARCH/g`

File: README.md (hunk 2 above)
Comment (tebbers) [reply README.md:89]: I guess I can't spell our own product name.

**Comment by benwis** [comment]
With the suggested changes, I have successfully connected erasearch  to the fly log shipper. Thanks!

**Comment by steveberryman** [comment]
This is great! If you could fix up the typo's mentioned I'll happily merge this!

**Comment by tebbers** [comment]
@steveberryman Fixed the spelling mistakes as well as added a transform to the config to make JSON more searchable.

**Comment by jsierles** [comment]
Hey, if we can get the conflicts resolved here, we can merge. Thanks.

**Comment by tebbers** [comment]
@jsierles conflicts resolved

**Comment by benwis** [comment]
@tebbers @jsierles Thanks for coming back and merging this!

**Pull request merged**